src/allotropy/parsers/agilent_gen5/agilent_gen5_structure.py

```diff
@@ -585,31 +723,57 @@ def create_metadata(header_data: HeaderData) -> Metadata:
         software_name=DEFAULT_SOFTWARE_NAME,
         software_version=header_data.software_version,
         file_name=header_data.file_name,
+        asm_file_identifier=NOT_APPLICABLE,
+        data_system_instance_id=NOT_APPLICABLE,
+        unc_path=NOT_APPLICABLE,
     )
 
 
+def _is_label_in_measurement_labels(label: str, measurement_labels: set[str]) -> bool:
+    if not measurement_labels:
+        return False
+    if label in measurement_labels:
+        return True
+    # TODO Improve this logic to support multiple kinetic sections
+    match = re.search(r"\[(\d+)]", label)
+    if match and match.group()[1:-1] in measurement_labels:
+        return True
+    return False
+
+
 def _create_measurement(
     measurement: MeasurementData,
     well_position: str,
     header_data: HeaderData,
     read_data: ReadData,
     sample_identifier: str | None,
     actual_temperature: float | None,
+    kinetic_data: KineticData | None = None,
+    kinetic_measurements: list[float | None] | None = None,
+    kinetic_elapsed_time: list[float] | None = None,
 ) -> Measurement:
-    # TODO(switch-statement): use switch statement once Benchling can use 3.10 syntax
-    if read_data.read_mode == ReadMode.ABSORBANCE:
-        measurement_type = MeasurementType.ULTRAVIOLET_ABSORBANCE
-    elif read_data.read_mode == ReadMode.FLUORESCENCE:
-        measurement_type = MeasurementType.FLUORESCENCE
-    elif read_data.read_mode == ReadMode.LUMINESCENCE:
-        measurement_type = MeasurementType.LUMINESCENCE
-
-    detector_wavelength_setting: JsonFloat | None = None
-    if measurement_type is MeasurementType.ULTRAVIOLET_ABSORBANCE:
+    match (read_data.read_mode, bool(kinetic_data)):
```

Comment:
Very sorry you went through changing this to a `match` statement, but we need to keep it `if` for now.

The problem is that Benchling can't use this syntax yet, not that it won't work here.

Reply:
ohhh I see, np I can revert.